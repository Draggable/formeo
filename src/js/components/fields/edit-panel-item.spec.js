import EditPanelItem from './edit-panel-item'

const mockField = {
  data: {
    attrs: {},
  },
  isDisabledProp: () => null,
  isLockedProp: () => null,
}

describe('EditPanelItem', () => {
  it('should match attribute item snapshot', () => {
    const item = new EditPanelItem({ key: 'attrs.type', data: { type: 'checkbox' }, field: mockField })
    expect(item).toMatchSnapshot()
  })
  it('should match option snapshot', () => {
    const itemData = { label: 'Checkbox 1', value: 'checkbox-1', checked: false }
    const item = new EditPanelItem({ key: 'options.0', data: itemData, index: 0, field: mockField })
    expect(item).toMatchSnapshot()
  })
})
