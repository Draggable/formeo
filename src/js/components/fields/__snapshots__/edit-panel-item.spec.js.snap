// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPanelItem should match attribute item snapshot 1`] = `
EditPanelItem {
  "conditionInput": [Function],
  "dom": <li
    class="field-attrs-type prop-wrap "
  >
    <div
      class="field-prop"
    >
      <div
        class="attrs-prop-inputs prop-inputs f-input-group"
      >
        <input
          class="type"
          id="attrs-type-type"
          name="type"
          placeholder="Type"
          type="text"
          value="checkbox"
        />
      </div>
      <div
        class="attrs-prop-controls prop-controls"
      >
        <button
          class="prop-remove prop-control"
          type="button"
        >
          remove
        </button>
      </div>
    </div>
  </li>,
  "field": Object {
    "data": Object {
      "attrs": Object {},
    },
    "isDisabledProp": [Function],
    "isLockedProp": [Function],
  },
  "generateConditionFields": [Function],
  "isDisabled": null,
  "isHidden": null,
  "isLocked": null,
  "itemFieldGroups": Array [],
  "itemIndex": undefined,
  "itemKey": "attrs.type",
  "itemValues": Array [
    Array [
      "type",
      "checkbox",
    ],
  ],
  "panelName": "attrs",
  "processConditionUIState": [Function],
}
`;

exports[`EditPanelItem should match option snapshot 1`] = `
EditPanelItem {
  "conditionInput": [Function],
  "dom": <li
    class="field-options-0 prop-wrap "
  >
    <div
      class="field-prop"
    >
      <div
        class="options-prop-inputs prop-inputs f-input-group"
      >
        <div
          class="f-addon"
        >
          <input
            class="checked"
            id="options-0-checked"
            name="checked[]"
            placeholder="Checked"
            type="checkbox"
          />
        </div>
        <input
          class="label"
          id="options-0-label"
          name="label"
          placeholder="Label"
          type="text"
          value="Checkbox 1"
        />
        <input
          class="value"
          id="options-0-value"
          name="value"
          placeholder="Value"
          type="text"
          value="checkbox-1"
        />
      </div>
      <div
        class="options-prop-controls prop-controls"
      >
        <button
          class="prop-remove prop-control"
          type="button"
        >
          remove
        </button>
      </div>
    </div>
  </li>,
  "field": Object {
    "data": Object {
      "attrs": Object {},
    },
    "isDisabledProp": [Function],
    "isLockedProp": [Function],
  },
  "generateConditionFields": [Function],
  "isDisabled": null,
  "isHidden": null,
  "isLocked": null,
  "itemFieldGroups": Array [],
  "itemIndex": 0,
  "itemKey": "options.0",
  "itemValues": Array [
    Array [
      "checked",
      false,
    ],
    Array [
      "label",
      "Checkbox 1",
    ],
    Array [
      "value",
      "checkbox-1",
    ],
  ],
  "panelName": "options",
  "processConditionUIState": [Function],
}
`;
